mParticle-Apple-SDK/mParticle.m

```diff
@@ -163,30 +163,47 @@ @implementation MPRokt
-        [queueParameters addParameter:nil];
+    NSArray<NSDictionary<NSString *, NSString *> *> *attributeMap = [self getRoktPlacementAttributes];
+    
+    // If attributeMap is nil the kit hasn't been initialized
+    if (attributeMap) {
+        NSMutableDictionary *mappedAttributes = attributes.mutableCopy;
+        for (NSDictionary<NSString *, NSString *> *map in attributeMap) {
+            NSString *mapFrom = map[@"map"];
+            NSString *mapTo = map[@"value"];
+            if (mappedAttributes[mapFrom]) {
+                NSString * value = mappedAttributes[mapFrom];
+                [mappedAttributes removeObjectForKey:mapFrom];
+                mappedAttributes[mapTo] = value;
+            }
+        }
+        for (NSString *key in mappedAttributes) {
+            [[MParticle sharedInstance].identity.currentUser setUserAttribute:key value:mappedAttributes[key]];
+        }
         
-        SEL roktSelector = @selector(executeWithViewName:attributes:placements:onLoad:onUnLoad:onShouldShowLoadingIndicator:onShouldHideLoadingIndicator:onEmbeddedSizeChange:filteredUser:);
-        [[MParticle sharedInstance].kitContainer_PRIVATE forwardSDKCall:roktSelector
-                                                                  event:nil
-                                                             parameters:queueParameters
-                                                            messageType:MPMessageTypeEvent
-                                                               userInfo:nil
-        ];
-    });
+        dispatch_async(dispatch_get_main_queue(), ^{
+            // Forwarding call to kits
+            MPForwardQueueParameters *queueParameters = [[MPForwardQueueParameters alloc] init];
+            [queueParameters addParameter:identifier];
+            [queueParameters addParameter:mappedAttributes];
+            [queueParameters addParameter:nil];
+            [queueParameters addParameter:nil];
+            [queueParameters addParameter:nil];
+            [queueParameters addParameter:nil];
+            [queueParameters addParameter:nil];
+            [queueParameters addParameter:nil];
+            
+            SEL roktSelector = @selector(executeWithViewName:attributes:placements:onLoad:onUnLoad:onShouldShowLoadingIndicator:onShouldHideLoadingIndicator:onEmbeddedSizeChange:filteredUser:);
+            [[MParticle sharedInstance].kitContainer_PRIVATE forwardSDKCall:roktSelector
+                                                                      event:nil
+                                                                 parameters:queueParameters
+                                                                messageType:MPMessageTypeEvent
+                                                                   userInfo:nil
+            ];
+        });
+    } else {
+        MPILogVerbose(@"[MParticle.Rokt selectPlacements:attributes:] not performed due to kit not being configured");
+    }
```

Comment:
This can also be DRYed up with the method below using all arguments and just pass either 2, or all arguments to it.